Docs/Book/Cookbook.rst

```diff
@@ -502,7 +555,71 @@ Create Fragments
    MolsToGridImage(Chem.GetMolFrags(rwmol, asMols=True))
 
 .. image:: images/RDKitCB_7_im6.png
+   :scale: 75%
+
+
+Sidechain-Core Enumeration 
+===========================
+
+| **Author:** Chris Earnshaw, Stephen Roughley, Greg Landrum (Vincent Scalfani added loop example)
+| **Source:** `<https://sourceforge.net/p/rdkit/mailman/message/35730514/>`_
+| **Index ID#:** RDKitCB_29
+| **Summary:** Replace sidechains on a core and enumerate the combinations.
+
+.. testcode::
+
+   from rdkit import Chem
+   from rdkit.Chem import Draw
+
+.. testcode::
+   
+   core = Chem.MolFromSmiles('*c1c(C)cccc1(O)')
+   chain = Chem.MolFromSmiles('CN*')
+   rxn = AllChem.ReactionFromSmarts('[c:1][#0].[#0][*:2]>>[c:1]-[*:2]')
+   ps = rxn.RunReactants((core,chain))
+   print(Chem.MolToSmiles(ps[0][0],isomericSmiles=True))
```

Comment:
I wonder if this leaves `ps` a little mysterious. Is `ps[0]` something worth printing out in some way? Can the internal structure of `ps` be hinted at in the recipe description?

Reply:
Maybe it would help to change the variable name to:

    product_lists = rxn.RunReactants((core,chain))
    print(Chem.MolToSmiles(product_lists[0][0]))

(also `isomericSmiles=True` is the default)

Reply:
What about:
```
descriptive_name = product_lists[0]
other_descriptive_name = descriptive_name[0]
print(Chem.MolToSmiles(other_descriptive_name))
```
where of course we choose actually descriptive names for the variables?

Reply:
Thanks for the fast review and feedback! I actually just found a similar example in the API docs, and I really like this format below. Or at least it makes the most sense to me. 

```
# core is '*c1c(C)cccc1(O)'
# chain is 'CN*'
rxn = AllChem.ReactionFromSmarts('[c:1][#0].[#0][*:2]>>[c:1]-[*:2]')
reacts = (Chem.MolFromSmiles('*c1c(C)cccc1(O)'),Chem.MolFromSmiles('CN*'))
products = rxn.RunReactants(reacts) # tuple
len(products)
```
1

`len(products[0])`

1

`print(Chem.MolToSmiles(products[0][0])) # [0][0] to index out the rdchem mol object`

'CNc1c(C)cccc1O'


I'll go ahead and update the PR with this in a few minutes, but I'm happy to revise again if you disagree.

Reply:
My thought was that it might be nice to pull `products[0]` and `products[0][0]` out into variables since that gives us an opportunity to name them before we print the second value. Whether that makes sense depends on whether the intermediate names would be obvious already to someone reading the cookbook. Not sure if that's so or not.